### PR TITLE
chore: Use git-cliff changelog generator

### PR DESCRIPTION
**Pull request by vbaranov**
## Motivation

Auto-generate changelog.

## Changelog

⚠️ devs attention ⚠️ 

- Multi-commits PR should have title starting with one of the prefix: `feat:`, `fix:`, `chore:`, `doc:`, `perf:`, `refactor:`.
- Single-commit PR should also have commit name starting from the same prefix.

On finalization of pull request, squashing option will be used (merge option is disabled in the repo).

Add [git-cliff](https://github.com/orhun/git-cliff) config to auto-generate changelog. It will be used manually per release issuing (since we cannot change existing commits to follow parseable structure).

- `- [ ] I added an entry to `CHANGELOG.md` with this PR` item is removed from Pull Request template.
- CONTRIBUTING.md is changed to reflect, that CHANGELOG is not required.

## Checklist for your Pull Request (PR)

  - [ ] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: cliff.toml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  { message = "^Refactor", group = "<!-- 2 -->🚜 Refactor" },
+  { message = "^chore\\(release\\): prepare for", skip = true },
+  { message = "^chore\\(deps.*\\)", skip = true },
+  { message = "^chore\\(pr\\)", skip = true },
+  { message = "^chore\\(pull\\)", skip = true },
+  { message = "^chore|^ci", group = "<!-- 7 -->⚙️ Miscellaneous Tasks" },
+  { message = "^Chore|^ci", group = "<!-- 7 -->⚙️ Miscellaneous Tasks" },
+  { body = ".*security", group = "<!-- 8 -->🛡️ Security" },
+  { message = "^revert", group = "<!-- 9 -->◀️ Revert" },
+]
```
Comment (sl1depengwyn) [inline cliff.toml:74]: We can do like this to make patterns case insensitive
```suggestion
commit_parsers = [
  { message = "(?i)^feat", group = "<!-- 0 -->🚀 Features" },
  { message = "(?i)^fix", group = "<!-- 1 -->🐛 Bug Fixes" },
  { message = "(?i)^doc", group = "<!-- 3 -->📚 Documentation" },
  { message = "(?i)^perf", group = "<!-- 4 -->⚡ Performance" },
  { message = "(?i)^refactor", group = "<!-- 2 -->🚜 Refactor" },
  { message = "(?i)^chore\\(release\\): prepare for", skip = true },
  { message = "(?i)^chore\\(deps.*\\)", skip = true },
  { message = "(?i)^chore\\(pr\\)", skip = true },
  { message = "(?i)^chore\\(pull\\)", skip = true },
  { message = "(?i)^chore|^ci", group = "<!-- 7 -->⚙️ Miscellaneous Tasks" },
  { body = ".*security", group = "<!-- 8 -->🛡️ Security" },
  { message = "^revert", group = "<!-- 9 -->◀️ Revert" },
]
```

File: cliff.toml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+# https://git-cliff.org/docs/configuration
+#
+# Lines starting with "#" are comments.
+# Configuration options are organized into tables and keys.
+# See documentation for more information on available options.
+
+[changelog]
+# changelog header
+header = """
+# ChangeLog\n
```
Comment (fedor-ivn) [inline cliff.toml:11]: It's not valuable, but just for the sake of aesthetics, I suggest we write the heading as follows:

```suggestion
# Changelog\n
```

This aligns with common practices as seen in the https://keepachangelog.com/en/1.1.0/ and other repos on GitHub

File: cliff.toml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    ## [{{ version | trim_start_matches(pat="v") }}] - {{ timestamp | date(format="%Y-%m-%d") }}
+{% else %}\
+    ## Current
+{% endif %}\
+{% for group, commits in commits | group_by(attribute="group") %}
+    ### {{ group | striptags | trim | upper_first }}
+    {% for commit in commits %}
+        - {% if commit.scope %}*({{ commit.scope }})* {% endif %}\
+            {% if commit.breaking %}[**breaking**] {% endif %}\
+            {{ commit.message | upper_first }}\
```
Comment (fedor-ivn) [inline cliff.toml:24]: Am I right that `upper_first` will capitalize the first word after a colon? For example, a commit like `feat: add some cool stuff` would appear in the Changelog as:

```md
# Features
- Add some cool stuff
```

Just curious if we should start the body of a commit message with a capital letter for consistency

**Review by k1rill-fedoseev — APPROVED** [review]

**Review by sl1depengwyn — APPROVED** [review]

**Review by nikitosing — APPROVED** [review]

File: cliff.toml (hunk 3 above)
Comment (vbaranov) [reply cliff.toml:24]: Yes, correct.

> Just curious if we should start the body of a commit message with a capital letter for consistency

There is no such requirement for now. This part was suggested by git-cliff. I'd prefer to leave it as is for now.

**Pull request merged**